### PR TITLE
* fix 632 percentile_doy

### LINKED ISSUE #632: Issue with percentile_doy
* Win 8.1 64-bit, Win 10 64-bit
* Python 3.8.3 64-bit
* numpy 1.18.4
* xarray 0.16.2
* dask 2021.1.1
* xclim 0.22.0

### Description
`percent_doy` function throws many warnings during call and takes what seems to be an excessive amount of time (>15 minutes) to run.

NC files for below example are available here: https://code.dented-planet.net/Indices.zip

#### Example code:
```
import xarray as xr
import xclim as xc
from xclim.core.calendar import percentile_doy

ds = xr.open_mfdataset(['tasmin.nc', 'tasmax.nc'])
tas = xc.indices.tas(ds.tasmin, ds.tasmax) 
t10 = percentile_doy(tas, per=0.1)
```
#### Repeated warnings during call:
```
C:\Users\C Drozdowski\AppData\Local\Programs\Python\Python38\lib\site-packages\dask\array\core.py:1428: FutureWarning: The `numpy.nanpercentile` function is not implemented by Dask array. You may want to use the da.map_blocks function or something similar to silence this warning. Your code may stop working in a future release.
```

### PR DESCRIPTION
**Pull request by tlogan2000**
### Pull Request Checklist:
- [X] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #632
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (major / minor / patch)` has been called on this branch
- [x] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Bug fix

* **Does this PR introduce a breaking change?** 
No

* **Other information**:

### CODE REVIEW

File: xclim/testing/tests/test_calendar.py [hunk 1]
```diff
@@ -76,6 +76,14 @@ def test_percentile_doy(tas_series):
     assert p1.sel(dayofyear=3, dim0=0).data == 2
     assert p1.attrs["units"] == "K"
 
+    # test with nan
```
Comment (Zeitsperre) [inline xclim/testing/tests/test_calendar.py:79]:
```suggestion
def test_percentile_doy_nan(tas_series):
```

**Review by Zeitsperre — APPROVED** [review]
I'd add a line in `internal changes` for the HISTORY about this. Looks good!

**Review by aulemahal — APPROVED** [review]
Nice!
Small but useful!

**Comment by Zeitsperre** [comment]
Don't forget to `bumpversion patch`!

**Pull request merged**